popmon/visualization/histogram_section.py

```diff
@@ -189,3 +217,131 @@ def _plot_histograms(feature, date, hc_list, hist_names):
+            hist_names.remove(" Heatmap ")
+        elif d == "row":
+            hist_names.remove(" Column Normalized Heatmap ")
+        else:
+            hist_names.remove(" Row Normalized Heatmap ")
+
+    # basic checks
+    if len(hist_names) <= 0:
+        # skip numeric heatmap
+        return {"plot": ""}
+
+    # filter out Nones (e.g. can happen with empty rolling hist)
+    none_hists = [i for i, hc in enumerate(hc_list) if hc is None]
+    hc_list = [hc for i, hc in enumerate(hc_list) if i not in none_hists]
+    hist_names = [hn for i, hn in enumerate(hist_names) if i not in none_hists]
+    # more basic checks
+    if len(hc_list) == 0:
+        return date, ""
+    assert_similar_hists(hc_list)
+    if hc_list[0].n_dim == 1:
+        props = get_hist_props(hc_list[0])
+        is_num = props["is_num"]
+        is_ts = props["is_ts"]
+        y_label = "Bin count" if len(hc_list) == 1 else "Bin probability"
+
+        if is_num:
+            # skip numeric heatmap
+            return {"plot": ""}
+        else:
+            # categorical, retrieve values and bins
+            entries_list, bins = get_consistent_numpy_entries(
+                hc_list, get_bin_labels=True
+            )  # bins = bin-labels
+
+            entries_list = np.stack(entries_list, axis=1)
+
+            # if cardinality of feature is more than 20
+            if len(bins) > top_n:
+                entries_list, bins = get_top_categories(entries_list, bins, top_n)
+
+            # make 3 copies : 1st normal, 2nd for column normalized heatmap, 3rd for row normalized heatmap
+            hists = []
+            if "normal" not in disable_heatmap:
+                hist_normal = entries_list.copy()
+                hists.append(hist_normal)
+
+            # normalize across column for a plot
+            if "column" not in disable_heatmap:
+                hist_col = entries_list.copy()
+                hist_col = np.stack(hist_col, axis=1)
+                hist_col = hist_col.astype(float)
+                for i in range(hist_col.shape[0]):
+                    if hist_col[i].sum() > 0:
+                        hist_col[i] = hist_col[i] / hist_col[i].sum()
+                hist_col = np.stack(hist_col, axis=1)
+                hists.append(hist_col)
+
+            # normalize across row for a plot
+            if "row" not in disable_heatmap:
+                hist_row = entries_list.copy()
+                hist_row = hist_row.astype(float)
+                for i in range(hist_row.shape[0]):
+                    if hist_row[i].sum() > 0:
+                        hist_row[i] = hist_row[i] / hist_row[i].sum()
+
+                hists.append(hist_row)
+
+        if len(bins) == 0:
+            # skip empty histograms
+            return {"plot": ""}
+
+        args = [
+            (hists[i], bins, date, feature, hist_names[i], y_label, is_num, is_ts, cmap)
+            for i in range(len(hists))
+        ]
+        plot = parallel(plot_heatmap_b64, args)
```

Comment:
Is the order returned by `parallel` guaranteed? Passing an identifier in `args` is probably safer